sklearn/neighbors/_unsupervised.py

```diff
@@ -105,7 +106,8 @@ class NearestNeighbors(KNeighborsMixin, RadiusNeighborsMixin,
     https://en.wikipedia.org/wiki/K-nearest_neighbor_algorithm
     """
 
-    def __init__(self, n_neighbors=5, radius=1.0,
+    @_deprecate_positional_args
+    def __init__(self, *, n_neighbors=5, radius=1.0,
```

Comment:
this one is a bit special because it has both n_neighbors and radius so I make both kwonly but feel free to disagree